pkg/csi_driver/controller.go

```diff
@@ -281,18 +281,27 @@ func (s *controllerServer) getCloudInstancesReservedIPRanges(ctx context.Context
 	if err != nil {
 		return nil, status.Error(codes.Aborted, err.Error())
 	}
+	// Due to unreachable location some instances may not show up here.
+	// TODO: create a new function to take a list of locations
+	// and return error if unreachable contained the region of interest.
 	multiShareInstances, err := s.config.fileService.ListMultishareInstances(ctx, &file.ListFilter{Project: filer.Project, Location: "-"})
 	if err != nil {
 		return nil, status.Error(codes.Aborted, err.Error())
 	}
 
-	// Initialize an empty reserved list. It will be populated with all the reservedIPRanges obtained from the cloud instances
+	// Initialize an empty reserved list. It will be populated with all the
+	// reservedIPRanges obtained from the cloud instances in the same VPC network
+	// as the ServiceInstance.
 	cloudInstancesReservedIPRanges := make(map[string]bool)
 	for _, instance := range instances {
-		cloudInstancesReservedIPRanges[instance.Network.ReservedIpRange] = true
+		if strings.EqualFold(instance.Network.Name, filer.Network.Name) {
```

Comment:
what would happen for default Network here? i.e. if the customer does not specify any network when creating instances, would the `filer.Network.Name` and `instance.Network.Name` still match up?

Reply:
If network is not specified, `default` will be stamped as the network value. 
See https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/pkg/csi_driver/controller.go#L453